### PR TITLE
Add link to view certificates for FA courses

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets? 
Part of #3233

#### What's this PR do?
Add a link to view a certificate for FA courses.

#### How should this be manually tested?
Create a passet passed course for your user:
```./manage.py alter_data set_past_run_to_passed --username annagav5 --program-title Digital --course-title 300```
If you need to create a certificate:
```
from courses.models import *
from grades.models import *
course_run = CourseRun.objects.get(edx_course_key=course_key)
grade=FinalGrade.objects.filter(course_run=course_run)[0]
cert=MicromastersCourseCertificate.objects.create(final_grade=grade)
```
screen shot 2017-08-24 at 3 03 24 pm

### CODE REVIEW

File: dashboard/api.py [hunk 1]
```diff
@@ -466,3 +467,22 @@ def has_to_pay_for_exam(mmtrack, course):
...
+    Args:
+        mmtrack (dashboard.utils.MMTrack): a instance of all user information about a program
+        course (courses.models.Course): A course
+    Returns:
+        str: url to view the certificate
+    """
+    final_grades = mmtrack.get_passing_final_grades_for_course(course)
+    url = ""
+    if final_grades.exists():
+        best_grade = final_grades.first()
```
Comment (amir-qayyum-khan) [inline dashboard/api.py]: @annagav are the grades sorted, best on first?

File: dashboard/utils.py [hunk 2]
```diff
@@ -298,6 +298,19 @@ def get_all_final_grades(self):
         )
         return {grade.course_run.edx_course_key: grade for grade in grades}
 
+    def get_passing_final_grades_for_course(self, course):
+        """
```
Comment (amir-qayyum-khan) [inline dashboard/utils.py]: please add a test where user has more the one grade and it returns best grade first.

File: dashboard/api.py [hunk 3]
```diff
@@ -466,3 +467,22 @@ def has_to_pay_for_exam(mmtrack, course):
     """
     attempt_limit = mmtrack.get_payments_count_for_course(course) * ATTEMPTS_PER_PAID_RUN
     return ExamAuthorization.objects.filter(user=mmtrack.user, course=course, exam_taken=True).count() >= attempt_limit
+
+
+def get_certificate_url(mmtrack, course):
```
Comment (amir-qayyum-khan) [inline dashboard/api.py]: Please add a test where user has more than one runs.

File: dashboard/api_test.py [hunk 4]
```diff
@@ -1628,3 +1665,32 @@ def test_has_to_pay(self, num_of_taken_exams, num_of_payments, result):
...
             ExamAuthorizationFactory.create(user=self.user, course=self.course, exam_taken=True)
         assert api.has_to_pay_for_exam(self.mmtrack, self.course) is result
+
+
+class GetCertificateForCourse(CourseTests):
+    """Tests get_certificate_url for a course"""
+
+    def setUp(self):
+        super().setUp()
+        self.mmtrack.user = self.user
```
Comment (amir-qayyum-khan) [inline dashboard/api_test.py]: as mentioned above add a test where user has more than passing runs

File: dashboard/api.py [hunk 5]
```diff
@@ -466,3 +467,22 @@ def has_to_pay_for_exam(mmtrack, course):
...
+    """
+    Find certificate associated with highest passing grade for the course
```
[the remark this answers is not in the document]
Comment (pdpinch) [reply dashboard/api.py]: Highest.

**Comment by amir-qayyum-khan** [comment]
can you tell me how to enable certificates?

**Comment by pdpinch** [comment]
@amir-qayyum-khan didn't you review https://github.com/mitodl/micromasters/pull/3490 ? The instructions are on that PR.

**Comment by amir-qayyum-khan** [comment]
@annagav where can i see certificate link?

screen shot 2017-08-25 at 8 39 05 pm

screen shot 2017-08-25 at 8 39 28 pm

**Comment by annagav** [comment]
@amir-qayyum-khan you need to use the management command `set_past_run_to_passed` I wrote in the description to make course in the past.

**Comment by amir-qayyum-khan** [comment]
i called that, should i called it like 

```docker-compose run web ./manage.py alter_data set_past_run_to_passed --username amirq --program-title Digital Learning --course-title Digital Learning 300```

or Digital 300 is enough,  i called it same a you mention (by just replacing my username)

**Comment by annagav** [comment]
I would try clearing all the cached enrollments for that user and that course.

**Comment by annagav** [comment]
You can try this:
```from dashboard.models import CachedEnrollment
CachedEnrollment.objects.filter(user=user, course_run=course_run).delete()
from grades.models import *
FinalGrade.objects.filter(user=user, course_run=course_run).delete()
```

**Comment by amir-qayyum-khan** [comment]
Found 2 issues

1) Should open certificate in new tab
screen shot 2017-08-25 at 10 58 12 pm

2) Certificate URL is wrong it is `http://192.168.99.101:8079/dashboard/certificate/0f0d0f144c93732869e1c7949c2e7368/`
remove dashboard from URL.

instead of `http://192.168.99.101:8079/certificate/0f0d0f144c93732869e1c7949c2e7368/`
screen shot 2017-08-25 at 10 58 18 pm

**Comment by annagav** [comment]
@amir-qayyum-khan, I addressed all your comments.

**Comment by amir-qayyum-khan** [comment]
please update tests

**Pull request merged**